app/modules/startup_optimizer.py

```
@@ -0,0 +1,143 @@
+            futures = {executor.submit(self.get_csv_data, year): year for year in years}
+            results = {}
+            for future in futures:
+                year = futures[future]
+                try:
+                    results[year] = future.result()
+                except Exception as e:
+                    print(f"Error loading CSV for year {year}: {e}")
+                    results[year] = pd.DataFrame()
+        return results
+
+    @lru_cache(maxsize=32)
```

Ruff (0.11.9)

Use of `functools.lru_cache` or `functools.cache` on methods can lead to memory leaks

(B019)